### PR TITLE
[KO] Translation of deepdream.ipynb

### PR DESCRIPTION
**Pull request by jaketae**
Uploaded Korean translation of tutorials/generative/deepdream.ipynb

### CODE REVIEW

**Comment by lamberta** [comment]
Thanks, @jaketae 
Please consider doing some reviews while you create new translations. The queue is pretty backed up now, which means it will take longer for this PR to get in. Thanks

**Comment by jaketae** [comment]
@lamberta Thank you for the reply! I totally understand the delay given that there are a lot of open pull requests at the moment. As a new contributor, it just didn't come to me that I could also review other submitted translations; I was under the impression that only the authorized reviewers mentioned by the tfdocsbot could. Is there anything that I should be particularly wary of when I go review other pull requests? Thanks in advance.

**Comment by lamberta** [comment]
Great. Thanks, @jaketae 

Check out the community translations section here: https://www.tensorflow.org/community/contribute/docs#community_translations
And you can communicate with the larger Korea ndocs community on the [docs-ko@tensorflow.org list](https://groups.google.com/a/tensorflow.org/forum/#!forum/docs-ko)

But it's pretty straightforward to add yourself as a reviewer, just send a PR with your github username added here: https://github.com/tensorflow/docs-l10n/blob/master/site/ko/REVIEWERS

Since these are community translations, it's up to you to organize for quality. From my perspective, I just need someone to look it over and make sure there's no spam, and in decent shape (ie. not just using Google Translate). I'm sure the Korean docs folks would be happy to have more reviewers willing to take a look :)

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "wgeIJg82NbG4"
+      },
+      "source": [
+        "## 몽환화(dream-ify)할 이미지 선택하기"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:175]: '몽환화' is hard to understand. Just keep it simple.
I suggest '몽환화할 이미지 선택하기' --> '변환할 이미지 선택하기'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 2]
```diff
@@ -0,0 +1,745 @@
...
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "Y5BPgc8NNbG0"
+      },
+      "outputs": [],
+      "source": [
+        "# 이미지를 내려받아 넘파이 어레이로 변환합니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:211]: '어레이' --> '배열'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 3]
```diff
@@ -0,0 +1,745 @@
...
+        "  if max_dim:\n",
+        "    img.thumbnail((max_dim, max_dim))\n",
+        "  return np.array(img)\n",
+        "\n",
+        "# 이미지를 정규화합니다.\n",
+        "def deprocess(img):\n",
+        "  img = 255*(img + 1.0)/2.0\n",
+        "  return tf.cast(img, tf.uint8)\n",
+        "\n",
+        "# 이미지를 진열합니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:225]: '진열' --> '출력'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 4]
```diff
@@ -0,0 +1,745 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "cruNQmMDNbGz"
+      },
+      "source": [
+        "사전 훈련된 이미지 분류 모델을 내려받습니다. 본 튜토리얼에서는 딥드림에서 원래 사용된 모델과 유사한 [InceptionV3](https://keras.io/applications/#inceptionv3)를 사용합니다. 다른 사전 학습된 모델을 사용해도 괜찮지만, 코드 구현 시 계층 이름을 수정해야한다는 점을 참고합니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:253]: For here and all other places in this notebook:
'계층' --> '층' 

'수정해야한다는 점을 참고합니다' --> '수정해야 합니다'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 5]
```diff
@@ -0,0 +1,745 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "Bujb0jPNNbGx"
+      },
+      "source": [
+        "딥드림은 활성화시킬 하나 혹은 그 이상의 계층을 선택한 후 \"손실\"을 최대화하도록 이미지를 수정함으로써 목표 계층을 \"흥분\"시키는 원리를 기반으로 합니다. 몽환화를 통해서 드러나는 특성의 복잡성은 선택한 계층에 따라 다르게 됩니다. 얕은 계층을 선택한다면 획 또는 간단한 패턴이 향상되고, 깊은 계층을 선택한다면 이미지 내의 복잡한 패턴이나 심지어 물체의 모습도 생성할 수 있습니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:276]: '목표 계층을' --> '선택한 층을'
'몽환화를 통해서 드러나는 특성의 복잡성은' --> '얼마나 복잡한 특성이 나타날지는'
'얕은 계층' --> '낮은 층' or '아래쪽 층'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 6]
```diff
@@ -0,0 +1,745 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "qOVmDO4LNbGv"
+      },
+      "source": [
+        "InceptionV3는 상당히 큰 모델입니다 (모델 구조의 그래프는 텐서플로 [연구 저장소](https://github.com/tensorflow/models/tree/master/research/inception)에서 확인할 수 있습니다). 많은 계층들 중, 딥드림을 구현하기 위해 살펴보아야 할 곳은 합성곱층들이 연결된 부분입니다. InceptionV3에는 'mixed0'부터 'mixed10'까지 총 11개의 이러한 합성곱층이 있습니다. 이 중 어떤 계층을 선택하느냐에 따라서 몽환화된 이미지의 모습이 결정됩니다. 깊은 계층은 눈이나 얼굴과 같은 고차원 특성(higher-level features)에 반응하는 반면, 얕은 계층은 선분이나 모양, 질감과 같은 저차원 특성에 반응합니다. 임의의 계층을 선택해 자유롭게 실험해보는 것도 가능합니다. 다만 깊은 계층(인덱스가 높은 계층)은 훈련을 위한 그래디언트 계산에 시간이 오래 걸릴 수 있음을 참고합니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:286]: '몽환화된 이미지의' --> '딥드림 이미지의'
'얕은 계층' --> '낮은 층' or '아래쪽 층'
'있음을 참고합니다' --> '있습니다'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 7]
```diff
@@ -0,0 +1,745 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "sb7u31B4NbGt"
+      },
+      "source": [
+        "## 손실 계산하기\n",
+        "\n",
+        "손실은 선택한 계층들의 활성화값의 총 합으로 계산됩니다. 계층의 크기와 상관 없이 모든 활성화값이 동일하게 고려될 수 있도록 각 계층의 손실을 정규화합니다. 일반적으로, 손실은 경상하강법으로 최소화하고자 하는 수치를 일컫습니다. 하지만 딥드림에서는 예외적으로 이 손실을 경사상승법(gradient ascent)을 통해 최대화할 것입니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:316]: '수치를 일컫습니다' --> '수치입니다'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 8]
```diff
@@ -0,0 +1,745 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "k4TCNsAUO9kI"
+      },
+      "source": [
+        "## 경사상승법\n",
+        "\n",
+        "선택한 계층의 손실을 구했다면, 이제 남은 순서는 입력 이미지에 대한 그래디언트를 계산하여 원본 이미지에 변형을 추가하는 것입니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:354]: '변형을 추가하는 것입니다' --> '추가하는 것입니다'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 9]
```diff
@@ -0,0 +1,745 @@
...
+        "\n",
+        "원본 이미지에 그래디언트를 더하는 것은 신경망이 보는 이미지 내의 패턴을 향상시키는 일에 해당합니다. 훈련이 진행될수록 신경망에서 선택한 계층을 더욱더 활성화시키는 이미지를 생성할 수 있습니다.\n",
+        "\n",
+        "이 작업을 수행하는 아래의 함수는 성능을 최적화하기 위해 [`tf.function`](https://www.tensorflow.org/api_docs/python/tf/function)에 포장(wrap)되었습니다. 이 함수는 `input_signature`를 이용해 함수가 다른 이미지 크기 혹은 `step`/`step_size`값에 대해 추적(trace)되지 않도록 합니다. 보다 자세한 설명을 위해서는 [Concrete functions guide](https://www.tensorflow.org/guide/concrete_function)를 참조합니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:358]: 'tf.function에 포장되었습니다' --> 'tf.function으로 감쌉니다' or 'tf.function 데코레이터를 적용합니다'

'추적되지 않도록 합니다' --> '트레이싱(tracing)되지 않도록 합니다'
If you are interested to tracing, check https://www.tensorflow.org/guide/concrete_function

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 10]
```diff
@@ -0,0 +1,745 @@
...
+        "        tf.TensorSpec(shape=[], dtype=tf.int32),\n",
+        "        tf.TensorSpec(shape=[], dtype=tf.float32),)\n",
+        "  )\n",
+        "  def __call__(self, img, steps, step_size):\n",
+        "      print(\"Tracing\")\n",
+        "      loss = tf.constant(0.0)\n",
+        "      for n in tf.range(steps):\n",
+        "        with tf.GradientTape() as tape:\n",
+        "          # `img`에 대한 그래디언트가 필요합니다.\n",
+        "          # `GradientTape`은 디폴트(default)로 오직 `tf.Variable`만 주시합니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:387]: '디폴트로' --> '기본적으로'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 11]
```diff
@@ -0,0 +1,745 @@
...
+        "id": "s9xqyeuwLZFy"
+      },
+      "source": [
+        "## 선택 사항: 타일(tile)을 이용해 이미지 확장하기\n",
+        "\n",
+        "한 가지 고려할 사항은 이미지의 크기가 커질수록 그래디언트 계산에 소요되는 시간과 메모리가 늘어난다는 점입니다. 따라서 위에서 구현한 방식은 옥타브 수가 많거나 입력 이미지의 해상도가 높은 상황에는 적합하지 않습니다.\n",
+        "\n",
+        "입력 이미지를 여러 타일로 나눠 각 타일에 대해 그래디언트를 계산하면 이 문제를 피할 수 있습니다.\n",
+        "\n",
+        "각 타일별 그래디언트를 계산하기 전에 이미지에 임의의 이동(random shift)를 적용시키면 타일 사이의 이음새가 나타나는 것을 방지할 수 있습니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:550]: '이미지에 임의의 이동를 적용시키면' --> '이미지를 랜덤하게 이동하면'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 12]
```diff
@@ -0,0 +1,745 @@
...
+        "\n",
+        "임의의 이동을 구현하는 작업부터 시작합니다:"
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:552]: '임의의 이동을 구현하는' --> '랜덤하게 이동시키는'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 13]
```diff
@@ -0,0 +1,745 @@
...
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "oGgLHk7o80ac"
+      },
+      "outputs": [],
+      "source": [
+        "def random_roll(img, maxroll):\n",
+        "  # 타일 경계선이 생기는 것을 방지하기 위해 이미지에 임의의 이동을 적용합니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:566]: '이미지에 임의의 이동을 적용합니다' --> '이미지를 랜덤하게 이동합니다'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 14]
```diff
@@ -0,0 +1,745 @@
...
+        "    ys = tf.range(0, img_rolled.shape[1], tile_size)[:-1]\n",
+        "    if not tf.cast(len(ys), bool):\n",
+        "      ys = tf.constant([0])\n",
+        "\n",
+        "    for x in xs:\n",
+        "      for y in ys:\n",
+        "        # 해당 타일의 그래디언트를 계산합니다.\n",
+        "        with tf.GradientTape() as tape:\n",
+        "          # `img_rolled`에 대한 그래디언트를 계산합니다.\n",
+        "          # ‘GradientTape`은 디폴트로 오직 `tf.Variable`만 주시합니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:635]: '디폴트로' --> '기본적으로'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 15]
```diff
@@ -0,0 +1,745 @@
...
+        "          tape.watch(img_rolled)\n",
+        "\n",
+        "          # 이미지에서 타일 하나를 추출합니다.\n",
+        "          img_tile = img_rolled[x:x+tile_size, y:y+tile_size]\n",
+        "          loss = calc_loss(img_tile, self.model)\n",
+        "\n",
+        "        # 해당 타일에 대한 이미지 그래디언트를 업데이트합니다.\n",
+        "        gradients = gradients + tape.gradient(loss, img_rolled)\n",
+        "\n",
+        "    # 이미지와 그래디언트에 실시한 이동을 취소하여 이미지를 원상 복구 시킵니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:645]: '실시한 이동을 취소하여 이미지를 원상 복구 시킵니다' --> '적용한 랜덤 이동을 취소합니다'

File: site/ko/tutorials/generative/deepdream.ipynb [hunk 16]
```diff
@@ -0,0 +1,745 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "0Og0-qLwNbGg"
+      },
+      "source": [
+        "지금까지 본 이미지보다 더 흥미로운 결과가 생성되었습니다! 이제는 옥타브의 수, 스케일, 그리고 활성화 계층을 달리해가며 이미지가 어떻게 몽환화되는지 실험해볼 차례입니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/deepdream.ipynb:739]: '이미지가 어떻게 몽환화되는지' --> '어떤 딥드림 이미지를 만들 수 있는지'

**Comment by jaketae** [comment]
My sincere apologies, I seemed to have messed up Git while applying a rebase and merge. I will close this PR in favor of a new one.

**Pull request closed without merging**